controllers/profiles/followAnonUser.js

```diff
@@ -0,0 +1,105 @@
+const moment = require('moment');
+
+const UserFollowUserFunction = require('../../databases/functions/userFollowUser');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const SuccessResponse = require('../../utils/response/SuccessResponse');
+
+const {UserFollowUser, UserFollowUserHistory, User, sequelize} = require('../../databases/models');
+const Getstream = require('../../vendor/getstream');
+const {addForFollowUser} = require('../../services/score');
+const BetterSocialCore = require('../../services/bettersocial');
+const UsersFunction = require('../../databases/functions/users');
+const {sendFollowMainFeedF2} = require('../../services/queue/mainFeedF2');
+
+module.exports = async (req, res) => {
+  const {user_id_followed, follow_source} = req.body;
+  const {user} = req;
+
+  if (req?.userId === user_id_followed)
+    return ErrorResponse.e403(res, 'Only allow following other profiles');
+
+  if (
+    await UserFollowUserFunction.checkIsUserFollowing(UserFollowUser, req?.userId, user_id_followed)
+  )
+    return ErrorResponse.e409(res, 'You have followed this user');
+
+  const targetUser = await UsersFunction.findUserById(User, user_id_followed);
+  if (!targetUser) {
+    return ErrorResponse.e404(res, 'User not found');
+  } else if (!targetUser.is_anonymous) {
+    return ErrorResponse.e403(res, 'Should be anonymous user');
+  }
+
+  try {
+    await sequelize.transaction(async (t) => {
+      const userFollowUser = await UserFollowUserFunction.registerAddFollowUser(
+        UserFollowUser,
+        UserFollowUserHistory,
+        req?.userId,
+        [user_id_followed],
+        follow_source,
+        t
+      );
+      return userFollowUser;
+    });
+  } catch (e) {
+    console.log('Error in follow anonymos user sql transaction');
+    return ErrorResponse.e409(res, e.message);
+  }
+
+  try {
+    const followUserExclusive = Getstream.feed.followUserExclusive(req?.userId, user_id_followed);
+    const followUser = Getstream.feed.followUser(req?.token, req?.userId, user_id_followed);
+    const followMainFeedFollowing = Getstream.feed.followMainFeedFollowing(
+      req?.token,
+      req?.userId,
+      user_id_followed
+    );
+    const sendJobF2User = sendFollowMainFeedF2(req.userId, user_id_followed);
+    // follow certain targeted feeds
+    // - exclusive
+    // - user
+    // - main_feed_following
+    await Promise.all([followUserExclusive, followUser, followMainFeedFollowing, sendJobF2User]);
+  } catch (e) {
+    console.log('Error in follow user v3 getstream');
+    console.log(e);
+    return ErrorResponse.e409(res, e.message);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code attempts to follow the target user's feeds and send a job to the F2 queue. If there's an error, it logs the error and returns a response with the error message. Again, consider logging a generic message instead of the error message directly.

```diff
  try {
    const followUserExclusive = Getstream.feed.followUserExclusive(req?.userId, user_id_followed);
    const followUser = Getstream.feed.followUser(req?.token, req?.userId, user_id_followed);
    const followMainFeedFollowing = Getstream.feed.followMainFeedFollowing(
      req?.token,
      req?.userId,
      user_id_followed
    );
    const sendJobF2User = sendFollowMainFeedF2(req.userId, user_id_followed);
    await Promise.all([followUserExclusive, followUser, followMainFeedFollowing, sendJobF2User]);
  } catch (e) {
-   console.log('Error in follow user v3 getstream');
+   console.error('Error in follow user v3 getstream', e);
    return ErrorResponse.e409(res, 'An error occurred while processing your request');
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->